### PR TITLE
Refactor history service config

### PR DESCRIPTION
**Pull request by wxing1292**
**What changed?**
* Move config into dedicated package

**Why?**
This PR provides better test coverage and code management.

**How did you test it?**
Run tests.

**Potential risks**
Minor business logic code change.

### CODE REVIEW

File: service/history/configs/config.go [hunk 1]
```diff
@@ -0,0 +1,429 @@
...
+	enumspb "go.temporal.io/api/enums/v1"
+
+	"go.temporal.io/server/common"
+	"go.temporal.io/server/common/definition"
+	"go.temporal.io/server/common/service/dynamicconfig"
+	"go.temporal.io/server/common/task"
+)
+
+// Config represents configuration for history service
+type Config struct {
```
Comment (samarabbas) [inline service/history/configs/config.go:39]: Will this cause linter rule violation?  I remember a rule which complains about stutter if the package and type names are similar.

File: service/history/service.go [hunk 2]
```diff
@@ -44,402 +42,17 @@ import (
 	"go.temporal.io/server/common/rpc"
 	"go.temporal.io/server/common/service/config"
 	"go.temporal.io/server/common/service/dynamicconfig"
-	"go.temporal.io/server/common/task"
+	"go.temporal.io/server/service/history/configs"
 )
 
-// Config represents configuration for history service
-type Config struct {
```
Comment (samarabbas) [inline service/history/service.go:51]: I'm expecting all configs are moved as is without any changes to new package.

File: service/history/configs/config.go (hunk 1 above)
Comment (wxing1292) [reply service/history/configs/config.go:39]: other than tons of `should have comment or be unexported` nothing really out of ordinary

File: service/history/service.go (hunk 2 above)
Comment (wxing1292) [reply service/history/service.go:51]: all changes are made / to be made as is, or at least no real business logic change for the purpose of refactoring

Comment (wxing1292) [reply service/history/service.go:51]: there can be go circular dependency, which requires some rewrite

**Review by alexshtin — APPROVED** [review]

File: service/history/configs/config.go (hunk 1 above)
Comment (alexshtin) [reply service/history/configs/config.go:39]: Someone need to fix all linter rules. I remember we were at the state when we had 0 violations.

**Pull request merged**